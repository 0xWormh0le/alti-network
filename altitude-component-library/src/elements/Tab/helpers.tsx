export const getTabIndexOf = (element: HTMLElement) => {
  const tabs = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tab') as HTMLElement[]
  return tabs.indexOf(element)
}

export const getTabPanelIndexOf = (element: HTMLElement) => {
  const tabPanels = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tabPanel') as HTMLElement[]
  return tabPanels.indexOf(element)
}
