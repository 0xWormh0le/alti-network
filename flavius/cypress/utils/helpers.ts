export const capitalize = (input: string) => {
  if (input.length > 0) {
    const output = input.toLowerCase().split('')
    output[0] = output[0].toUpperCase()
    return output.join('')
  }
  return input
}
