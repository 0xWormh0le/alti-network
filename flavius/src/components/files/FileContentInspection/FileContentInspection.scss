@import 'styles/_shared.scss';

$MAX_WIDTH: 360px;
$LABEL_WIDTH: 150px;

.FileContentInspectionTableWrapper {
  width: $MAX_WIDTH;

  &--Phrase {
    .Table__heading-cell:nth-child(2) {
      text-align: center;
    }
  }

  th:nth-child(1) {
    padding-left: 0;
  }

  tr {
    td .center {
      text-align: center;
    }
  }

  &__loader {
    position: relative;
    margin-bottom: 10px;
    margin-top: 10px;
  }

  &__no-results {
    display: flex;
    height: 20em;
    align-items: center;
    justify-content: center;
    padding: $BASE_SPACING * 5;
    border-top: 1px solid $BORDER_COLOR;

    &-text {
      flex: 1;
      text-align: center;
      border-radius: 4px;
      border: 1px dashed $BORDER_COLOR;
      padding: $BASE_SPACING * 3;
      color: $PLACEHOLDER_COLOR;
    }
  }

  &__keywordLabel {
    max-width: $LABEL_WIDTH;
    height: 23px;
    padding: 0px 6px;
    border-radius: 4px;
    border: solid 1px $LIGHT_GREY_MEDIUM;
    background-color: #f6f6f6;
    @include set-typography('SUBHEAD2', 'bold');
    letter-spacing: normal;
    text-align: left;
    color: $TOMATO;
    font-family: $FONT_FAMILY_COURIER;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    display: table-caption;
  }
}
