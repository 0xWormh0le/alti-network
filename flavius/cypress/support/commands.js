import * as Sentry from '@sentry/browser'
import { RISK_TYPE_IDS } from '../utils/config'
const { MANY_DOWNLOADS_BY_APP_ID } = RISK_TYPE_IDS
import { version } from '../../package.json'

Cypress.on('fail', (error, runnable) => {
  console.info('we failed!!', error, runnable)
  const errorMessage = `Heartbeat Error detected in spec '${runnable.parent.title}': ${error.message}`
  Sentry.init({
    dsn: 'https://4345e62a3abc48049f9965ccc17ed310@sentry.io/1306610',
    environment: Cypress.env('REACT_APP_STAGE'),
    release: `flavius@${version}`,
    integrations: [
      new Sentry.Integrations.Breadcrumbs({
        console: Cypress.env('REACT_APP_STAGE') === 'production',
        message: errorMessage
      })
    ]
  })
  const err = new Error(errorMessage)
  Sentry.withScope((scope) => {
    scope.setExtra('stacktrace', error.stack)
    scope.setExtra('codeframe', error.codeFrame)
    Sentry.captureException(err)
  })
  throw err
})

const checkMinimumValidPerson = (person) => {
  if (person) {
    expect(person.accessCount).to.be.a('number')
    expect(person.primaryEmail?.address).to.be.a('string')
    expect(person.hasOwnProperty('name')).to.be.true
  }
}

const checkMinimumFile = (file) => {
  const {
    createdAt,
    createdBy,
    externalAccessCount,
    externalAccessList,
    fileId,
    fileName,
    internalAccessCount,
    internalAccessList,
    lastModified,
    linkVisibility,
    lastIngested,
    iconUrl
  } = file
  if (createdAt) {
    expect(createdAt).to.be.a('number')
  }
  if (iconUrl) {
    expect(iconUrl).to.be.a('string')
  }
  expect(externalAccessCount).to.be.a('number')
  expect(externalAccessList).to.be.an('Array')
  // todo: uncomment. id and name is currently null
  expect(fileId).to.be.a('string')
  expect(fileName).to.be.a('string')
  expect(internalAccessCount).to.be.a('number')
  expect(internalAccessList).to.be.an('Array')
  if (lastModified) {
    expect(lastModified).to.be.a('number')
  }
  if (lastIngested) {
    expect(lastIngested).to.be.a('number')
  }
  expect(linkVisibility).to.be.a('string')
}

const checkMinimumFiles = (files) => checkMinimumFile(files[randomIndex(files)])

const randomIndex = (array) => Math.floor(Math.random() * array.length)

Cypress.Commands.add('login', () => {
  cy.get('#email').type('bobbie@thoughtlabs.io')
  cy.get('#password').type(Cypress.env('CYPRESS_USER_PASSWORD'))
  cy.get('.Login__button').click()
})

Cypress.Commands.add('checkMinimumValidPerson', checkMinimumValidPerson)

Cypress.Commands.add('checkFullPerson', (person) => {
  checkMinimumValidPerson(person)
  expect(person.externalCount).to.be.a('number')
  expect(person.userKind).to.be.a('number')
  expect(person.internalCount).to.be.a('number')
})

Cypress.Commands.add('checkFiles', (files) => {
  expect(files).to.be.an('Array')
  checkMinimumFiles(files)
})

Cypress.Commands.add('checkRisks', (risks, filteredRiskTypeId) => {
  expect(risks).to.be.an('Array')
  const randomRiskIndex = randomIndex(risks)
  const randomTestRisk = risks[randomRiskIndex]
  expect(randomTestRisk.platformId).to.be.a('string')
  expect(randomTestRisk.datetime).to.be.a('number')
  expect(randomTestRisk.fileCount).to.be.a('number')
  expect(randomTestRisk.hasOwnProperty('fileId')).to.be.true
  expect(randomTestRisk.hasOwnProperty('fileName')).to.be.true
  expect(randomTestRisk.riskDescription).to.be.a('string')
  expect(randomTestRisk.riskId).to.be.a('string')
  expect(randomTestRisk.riskTarget).to.be.an('Array')
  if (filteredRiskTypeId) {
    expect(randomTestRisk.riskTypeId).to.equal(filteredRiskTypeId)
  }
  expect(randomTestRisk.severity).to.be.a('number')

  checkMinimumValidPerson(randomTestRisk.creator)
  checkMinimumValidPerson(randomTestRisk.owner)

  if (filteredRiskTypeId === MANY_DOWNLOADS_BY_APP_ID) {
    expect(randomTestRisk.plugin.id).to.be.a('string')
    // todo: uncomment. name is currently null
    expect(randomTestRisk.plugin.name).to.be.a('string')
  }
})

Cypress.Commands.add('checkPagination', (paginationResponse, overrides, extra) => {
  expect(paginationResponse.orderBy).to.equal(
    overrides && overrides.hasOwnProperty('orderBy') ? overrides.orderBy : 'datetime'
  )
  expect(paginationResponse.pageCount).to.be.a('number')
  expect(paginationResponse.pageNumber).to.be.a('number')
  expect(paginationResponse.pageSize).to.be.a('number')
  expect(paginationResponse.sort).to.equal(overrides && overrides.hasOwnProperty('sort') ? overrides.sort : 'DESC')

  if (extra) {
    const { wrapper } = extra
    const { pageCount } = paginationResponse
    if (pageCount > 1) {
      if (pageCount > 5) {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', 5)
      } else {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', pageCount)
      }
    }
  }
})

Cypress.Commands.add('checkEvents', (events, filesIsObject) => {
  expect(events).to.be.an('Array')
  const randomEventIndex = randomIndex(events)
  const randomTestEvent = events[randomEventIndex]
  expect(randomTestEvent.datetime).to.be.a('number')
  expect(randomTestEvent.eventDescription).to.be.a('string')
  expect(randomTestEvent.eventId).to.be.a('string')
  expect(randomTestEvent.eventName).to.be.a('string')
  expect(randomTestEvent.targetPeople).to.be.an('Array')
  if (!filesIsObject) {
    expect(randomTestEvent.files).to.be.an('Array')
    const randomFileIndex = randomIndex(randomTestEvent.files)
    checkMinimumFile(randomTestEvent.files[randomFileIndex])
  } else {
    checkMinimumFiles(randomTestEvent.files)
  }
})
