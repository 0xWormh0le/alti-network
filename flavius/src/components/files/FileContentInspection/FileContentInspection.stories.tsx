import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import FileContentInspection, { FileContentInspectionProps } from './FileContentInspection'

export default {
  title: 'Files/FileContentInspection',
  component: FileContentInspection
} as Meta

const Template: Story<FileContentInspectionProps> = (args) => <FileContentInspection {...args} />

export const Default = Template.bind({})
Default.args = {
  ccn: 1,
  sensitiveKeywords: [
    {
      keyword: 'someSensitiveKeyword',
      count: 1
    }
  ],
  ssn: 1
}

export const Loading = Template.bind({})
Loading.args = {
  ccn: 1,
  sensitiveKeywords: [
    {
      keyword: 'someSensitiveKeyword',
      count: 1
    }
  ],
  ssn: 1,
  loading: true
}

export const OnlySensitivePhrases = Template.bind({})
OnlySensitivePhrases.args = {
  sensitiveKeywords: [
    {
      keyword: 'someSensitiveKeyword',
      count: 1
    },
    {
      keyword: 'otherSensitiveWord',
      count: 10
    }
  ]
}
