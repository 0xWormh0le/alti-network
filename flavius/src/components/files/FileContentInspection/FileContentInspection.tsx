import React, { useEffect, useMemo, useState } from 'react'
import ContentLoader from 'react-content-loader'
import FileSaver from 'file-saver'
import { ModernTable } from '@altitudenetworks/component-library'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import EntityCountPagination from 'components/elements/EntityCountPagination'
import Checkmark from 'icons/checkmark'
import { csvFileName, pluralize, renderAttributeIfDev } from 'util/helpers'
import UI_STRINGS from 'util/ui-strings'
import './FileContentInspection.scss'

export interface FileContentInspectionProps {
  sensitiveKeywords: PhraseObject[]
  ccn: number
  ssn: number
  loading?: boolean
}

export interface FileContentPatternsProps {
  ccn: number
  ssn: number
}

export interface FileContentInspectionPhrase {
  count: number
  keyword: string
}

export interface FileContentInspectionPhrases {
  sensitiveKeywords?: FileContentInspectionPhrase[]
}

export interface PhraseObject {
  count: number
  keyword: string
}

export interface FileContentSensitivePhrasesPaginationProps {
  pageCount: number
  currentPage: number
  itemCount: number
  onPageChange: (page: number) => void
  sensitiveKeywords: []
}

const PAGE_SIZE = 5

const FileContentInspection: React.FC<FileContentInspectionProps> = ({ sensitiveKeywords, ccn, ssn, loading }) => {
  if (loading) {
    return <TableLoading />
  } else if (sensitiveKeywords) {
    return <FileContentInspectionComp sensitiveKeywords={sensitiveKeywords} ccn={ccn} ssn={ssn} />
  } else {
    return (
      <div className='FileContentInspectionTableWrapper__no-results'>
        <Typography
          variant={TypographyVariant.BODY_LARGE}
          className='FileContentInspectionTableWrapper__no-results-text'>
          {UI_STRINGS.SPOTLIGHT.NO_RESULT_FOUND}
        </Typography>
      </div>
    )
  }
}

export const FileContentInspectionComp: React.FC<FileContentInspectionProps> = ({
  sensitiveKeywords,
  ccn,
  ssn,
  loading
}) => {
  const [showPatterns, setShowPatterns] = useState(false)
  const [showSensitiveKeywords, setShowSensitiveKeywords] = useState(false)
  const [currentPage, setCurrentPage] = useState(0)

  const pageCount = Math.ceil(sensitiveKeywords.length / PAGE_SIZE)

  const itemsInPage = useMemo(
    () =>
      sensitiveKeywords.filter(
        (item, index) => index >= (currentPage - 1) * PAGE_SIZE && index < currentPage * PAGE_SIZE
      ),
    [sensitiveKeywords, currentPage]
  )

  useEffect(() => {
    if (sensitiveKeywords) {
      setShowPatterns(ccn > 0 || ssn > 0)
      if (sensitiveKeywords.length) {
        setCurrentPage(1)
        setShowSensitiveKeywords(true)
      }
    }
  }, [sensitiveKeywords, ccn, ssn])

  return (
    <React.Fragment>
      {showPatterns && (
        <ModernTable
          fields={[UI_STRINGS.FILE.PATTERNS, UI_STRINGS.FILE.RESULTS]}
          className='FileContentInspectionTableWrapper'
          items={[
            { type: 'ssn', value: ssn },
            { type: 'ccn', value: ccn }
          ].filter((data) => data.value > 0)}
          scopedSlots={{
            [UI_STRINGS.FILE.PATTERNS]: (val) => (
              <div>{val.type === 'ccn' ? UI_STRINGS.RISKS.SENSITIVE_CCN : UI_STRINGS.RISKS.SENSITIVE_SSN}</div>
            ),
            [UI_STRINGS.FILE.RESULTS]: (val) =>
              val.type === 'ccn' ? (
                <div {...renderAttributeIfDev({ 'data-testid': 'ccnCount' })}>
                  {ccn} {pluralize('instance', val.value)} found
                </div>
              ) : (
                <div {...renderAttributeIfDev({ 'data-testid': 'ssnCount' })}>
                  {ssn} {pluralize('instance', val.value)} found
                </div>
              )
          }}
        />
      )}
      {showSensitiveKeywords && (
        <ModernTable
          fields={[UI_STRINGS.FILE.PHRASE, UI_STRINGS.FILE.EXACT_MATCH, UI_STRINGS.FILE.RESULTS]}
          className='FileContentInspectionTableWrapper FileContentInspectionTableWrapper--Phrase'
          items={itemsInPage}
          scopedSlots={{
            [UI_STRINGS.FILE.PHRASE]: (val: PhraseObject) => (
              <div
                className='FileContentInspectionTableWrapper__keywordLabel'
                {...renderAttributeIfDev({
                  'data-testid': `keyword-${itemsInPage.findIndex((item) => item.keyword === val.keyword)}`
                })}>
                {val.keyword}
              </div>
            ),
            [UI_STRINGS.FILE.EXACT_MATCH]: () => (
              <div className='center'>
                <Checkmark className='PhraseRow__icon' />
              </div>
            ),
            [UI_STRINGS.FILE.RESULTS]: (val: PhraseObject) => (
              <div>
                {val.count} {pluralize('instance', val.count)}
              </div>
            )
          }}>
          <EntityCountPagination
            onPageChange={setCurrentPage}
            entityCount={sensitiveKeywords.length}
            onExportCsv={() => onExportCSV(sensitiveKeywords as [])}
            pageNumber={currentPage}
            pageSize={PAGE_SIZE}
            pageCount={pageCount}
            hidePageNumbers={true}
          />
        </ModernTable>
      )}
    </React.Fragment>
  )
}

const onExportCSV = (sensitiveKeywords: []) => {
  const docTitle = csvFileName('Altitude_Export_Sensitive_Phrases')
  type IMapResponseToCSV = (_csvData: TableDisplay[]) => string
  const mapResponseToCSV: IMapResponseToCSV = (csvData: TableDisplay[]) => {
    let returnString = `Phrase,Exact Match,Results\n`
    csvData.forEach((item: any) => {
      returnString += `${item.keyword},Exact Match,${item.count}\n`
    })
    return returnString
  }
  const blob = new Blob([mapResponseToCSV(sensitiveKeywords)], { type: 'data:text/csv;charset=utf-8,' })
  FileSaver.saveAs(blob, docTitle)
}

const TableLoading = () => (
  <div className='FileContentInspectionTableWrapper__loader'>
    <ContentLoader
      backgroundColor='#f0f0f0'
      foregroundColor='#f7f7f7'
      width={360}
      height={90}
      className='ChartLoading'
      uniqueKey='ChartLoading'>
      <rect x='0' y='0' width='360' height='90' />
    </ContentLoader>
  </div>
)

export default FileContentInspection
