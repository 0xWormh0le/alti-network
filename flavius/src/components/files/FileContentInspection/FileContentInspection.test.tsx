import React from 'react'
import { renderWithRouter } from 'test/support/helpers'
import FileContentInspection from './FileContentInspection'
import { FileContentInspectionData } from 'test/mocks'
import { act, waitForElement, render } from '@testing-library/react'

const props: any = {
  sensitiveKeywords: FileContentInspectionData.propsWithRisks.sensitivePhrases.sensitiveKeywords,
  ccn: FileContentInspectionData.propsWithRisks.sensitivePhrases.ccNum,
  ssn: FileContentInspectionData.propsWithRisks.sensitivePhrases.ssn,
  loading: false
}

describe('SensitiveFileCell', () => {
  it('renders correctly', () => {
    const { container } = renderWithRouter(<FileContentInspection {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('renders a ccn count of 6', async () => {
    const { getByTestId } = render(<FileContentInspection {...props} />)
    await act(async () => {
      const element = await waitForElement(() => getByTestId('ccnCount'))
      expect(element).toHaveTextContent('6 instances found')
    })
  })

  it('renders sensitivekeyword "merit"', async () => {
    const { getByTestId } = render(<FileContentInspection {...props} />)
    await act(async () => {
      const element = await waitForElement(() => getByTestId('keyword-0'))
      expect(element).toHaveTextContent('merit')
    })
  })
})
