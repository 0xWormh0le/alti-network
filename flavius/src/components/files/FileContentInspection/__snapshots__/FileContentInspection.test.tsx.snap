// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SensitiveFileCell renders correctly 1`] = `
<div>
  <div
    class="Table FileContentInspectionTableWrapper"
  >
    <table
      class="Table__table"
    >
      <thead>
        <tr
          class="Table__heading"
        >
          <th
            class="Table__heading-cell"
          >
            <span>
              patterns
            </span>
          </th>
          <th
            class="Table__heading-cell"
          >
            <span>
              results
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="Table__row"
        >
          <td
            class="Table__cell"
          >
            <div>
              Social Security Number
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div
              data-testid="ssnCount"
            >
              3
               
              instances
               found
            </div>
          </td>
        </tr>
        <tr
          class="Table__row"
        >
          <td
            class="Table__cell"
          >
            <div>
              Credit Card Number
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div
              data-testid="ccnCount"
            >
              6
               
              instances
               found
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="Table FileContentInspectionTableWrapper FileContentInspectionTableWrapper--Phrase"
  >
    <table
      class="Table__table"
    >
      <thead>
        <tr
          class="Table__heading"
        >
          <th
            class="Table__heading-cell"
          >
            <span>
              phrase
            </span>
          </th>
          <th
            class="Table__heading-cell"
          >
            <span>
              exact match
            </span>
          </th>
          <th
            class="Table__heading-cell"
          >
            <span>
              results
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="Table__row"
        >
          <td
            class="Table__cell"
          >
            <div
              class="FileContentInspectionTableWrapper__keywordLabel"
              data-testid="keyword-0"
            >
              merit
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div
              class="center"
            >
              <svg
                class="PhraseRow__icon"
                height="9"
                viewBox="0 0 13 9"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.188 1.824L5.452 8.687a1.06 1.06 0 0 1-1.506 0L.812 5.541a1.072 1.072 0 0 1 0-1.511 1.062 1.062 0 0 1 1.506 0l2.381 2.39L10.682.313a1.062 1.062 0 0 1 1.506 0 1.071 1.071 0 0 1 0 1.511z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div>
              4
               
              instances
            </div>
          </td>
        </tr>
        <tr
          class="Table__row"
        >
          <td
            class="Table__cell"
          >
            <div
              class="FileContentInspectionTableWrapper__keywordLabel"
              data-testid="keyword-1"
            >
              acme
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div
              class="center"
            >
              <svg
                class="PhraseRow__icon"
                height="9"
                viewBox="0 0 13 9"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.188 1.824L5.452 8.687a1.06 1.06 0 0 1-1.506 0L.812 5.541a1.072 1.072 0 0 1 0-1.511 1.062 1.062 0 0 1 1.506 0l2.381 2.39L10.682.313a1.062 1.062 0 0 1 1.506 0 1.071 1.071 0 0 1 0 1.511z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div>
              3
               
              instances
            </div>
          </td>
        </tr>
        <tr
          class="Table__row"
        >
          <td
            class="Table__cell"
          >
            <div
              class="FileContentInspectionTableWrapper__keywordLabel"
              data-testid="keyword-2"
            >
              drainage
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div
              class="center"
            >
              <svg
                class="PhraseRow__icon"
                height="9"
                viewBox="0 0 13 9"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.188 1.824L5.452 8.687a1.06 1.06 0 0 1-1.506 0L.812 5.541a1.072 1.072 0 0 1 0-1.511 1.062 1.062 0 0 1 1.506 0l2.381 2.39L10.682.313a1.062 1.062 0 0 1 1.506 0 1.071 1.071 0 0 1 0 1.511z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div>
              2
               
              instances
            </div>
          </td>
        </tr>
        <tr
          class="Table__row"
        >
          <td
            class="Table__cell"
          >
            <div
              class="FileContentInspectionTableWrapper__keywordLabel"
              data-testid="keyword-3"
            >
              chopper
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div
              class="center"
            >
              <svg
                class="PhraseRow__icon"
                height="9"
                viewBox="0 0 13 9"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.188 1.824L5.452 8.687a1.06 1.06 0 0 1-1.506 0L.812 5.541a1.072 1.072 0 0 1 0-1.511 1.062 1.062 0 0 1 1.506 0l2.381 2.39L10.682.313a1.062 1.062 0 0 1 1.506 0 1.071 1.071 0 0 1 0 1.511z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div>
              5
               
              instances
            </div>
          </td>
        </tr>
        <tr
          class="Table__row"
        >
          <td
            class="Table__cell"
          >
            <div
              class="FileContentInspectionTableWrapper__keywordLabel"
              data-testid="keyword-4"
            >
              erase
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div
              class="center"
            >
              <svg
                class="PhraseRow__icon"
                height="9"
                viewBox="0 0 13 9"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.188 1.824L5.452 8.687a1.06 1.06 0 0 1-1.506 0L.812 5.541a1.072 1.072 0 0 1 0-1.511 1.062 1.062 0 0 1 1.506 0l2.381 2.39L10.682.313a1.062 1.062 0 0 1 1.506 0 1.071 1.071 0 0 1 0 1.511z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </td>
          <td
            class="Table__cell"
          >
            <div>
              1
               
              instance
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="Table__children"
    >
      <div
        class="EntityCountPagination"
      >
        <div>
          <button
            class="Button EntityCountPagination__export-button Button--secondary"
            type="button"
          >
            Export All
          </button>
        </div>
        <div
          class="EntityCountPagination__pagination-info"
        >
          <div
            class="EntityCountPagination__page-size EntityCountPagination__items-count"
          >
            <span>
              Displaying 5 Items Per Page
            </span>
          </div>
          <div
            class="Pagination"
          >
            <div
              class="Pagination__group"
            >
              <div
                aria-label="prev"
                class="PaginationItem PaginationItem--arrow"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link PaginationItem__link--disabled"
                >
                  <svg
                    class="Pagination__arrow"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polygon
                      fill="currentColor"
                      points="5,10 15,1 15,19"
                    />
                  </svg>
                </button>
              </div>
              <div
                aria-label="next"
                class="PaginationItem PaginationItem--arrow"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  <svg
                    class="Pagination__arrow"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polygon
                      fill="currentColor"
                      points="15,10 5,1 5,19"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
